### PR TITLE
Fix NuGet packages.config parsing

### PR DESCRIPTION
**Pull request by brandonaut**
Since pattern whitespace is being ignored via the `x` flag, the string `<packages>` incorrectly matches the second part of the regex (`<package\s[^>]*?[^/]>.*?</package>`) when "package" isn't a self-closing element.

For example:

```xml
<packages>
  <package id="mypackage" version="0.0.0"></package>
</packages>
```

will match:

```
<packages>
  <package id="mypackage" version="0.0.0"></package>
```

This commit makes the space explicit so that `<packages>` no longer matches.

### CODE REVIEW

**Comment by jurre** [comment]
Makes sense to me, thanks! Would you mind adding a test to verify the new behavior and to make sure we don't regress in the future?

**Comment by brandonaut** [comment]
Working on it! I'm not very familiar with Ruby and RSpec, so it's taking me a bit...

**Comment by jeffwidman** [comment]
I approved the CI run, let us know when it's ready for another round of review or if you need CI runs approved again.

**Comment by brandonaut** [comment]
I'm ready for another CI run. Sorry, I'm not set up to run the tests locally :(

**Comment by jeffwidman** [comment]
No problem at all, just approved. I also for now changed our CI rules so that every run from a first-time contributor doesn't require pre-approval, but only if they're new to GitHub. That should make your life easier because you won't have to wait for us to approve anymore, sorry for the hassle.

Also for running tests locally it's actually super simple since you can do it directly within docker... see https://github.com/dependabot/dependabot-core#running-with-docker and then also see the "tests" section slightly down. Within a failing test you can drop a `binding.b` at any point to drop yourself into a debugger.

Holler if you have questions.

File: nuget/spec/fixtures/packages_configs/packages.config
Comment (jeffwidman) [inline nuget/spec/fixtures/packages_configs/packages.config:11]: I assume this test fails w/o this patch?

**Review by jeffwidman — APPROVED** [review]
👍 nice job thank you!

File: nuget/spec/fixtures/packages_configs/packages.config
Comment (brandonaut) [reply nuget/spec/fixtures/packages_configs/packages.config:11]: Correct

Comment (brandonaut) [reply nuget/spec/fixtures/packages_configs/packages.config:11]: I got the tests running locally and was able to confirm :)

**Pull request merged**
